### ISSUE TITLE
Ridge in "forward selection" and "highest weight"

### ISSUE THREAD

**Issue by beopis** [thomasp85/lime#134]
Thank you for your work!
My question is: why `select_f_fs()` and `select_f_hw()`, used in the model selection step, have fixed `lambda = 0` in `glmnet()`? 
From the help of `glmnet()`, we see that the loss function of a GLMnet gaussian model is
![lossfunct1](https://user-images.githubusercontent.com/39793891/49102776-fcbcdb00-f279-11e8-9440-492c78782734.png)
(where P is the penalty term); doesn't it mean that these model selections are using a Ordinary Least Squares regression instead of a ridge regression, as reported in `lime`'s documentation?

**Comment by beopis** [comment]
> Thank you for your work!
> My question is: why `select_f_fs()` and `select_f_hw()`, used in the model selection step, have fixed `lambda = 0` in `glmnet()`?
> From the help of `glmnet()`, we see that the loss function of a GLMnet gaussian model is
> ![lossfunct1](https://user-images.githubusercontent.com/39793891/49102776-fcbcdb00-f279-11e8-9440-492c78782734.png)
> (where P is the penalty term); doesn't it mean that these model selections are using a Ordinary Least Squares regression instead of a ridge regression, as reported in `lime`'s documentation?

**Comment by beopis** [comment]
@thomasp85, did I misunderstood the code ... ?

**Comment by iliasp23** [comment]
When you call glmnet function, a=0 stands for ridge regression and a=1 for lasso regression.
Both of them, are using ordinary least squares. The difference is that they have a tuning parameter (λ) which penalizes the least squares results in order to achieve a more parsimonious model and deal with overfitting.  

If you choose lasso regression, the penalty is more "strict" comparing with ridge's penalty. So the lasso regression can be used for model selection because it completely removes the influence of a variable that is not needed. But if you choose ridge regression the final model will have all the variables but their coefficients will be shrunk towards 0.

**Comment by beopis** [comment]
This doesn't answer my question ... my question is why is it fixed lambda=0 in the code?  Doing this, the penalty term has no effect, and the model selection is not RIDGE, but OLS.

**Comment by thomasp85** [comment]
This is probably due to my lack of understanding of `lambda` when I wrote it.. will get fixed in next release

**Comment by thomasp85** [comment]
Ok, so I guess this will not change. The choice of `lambda = 0` is a result of the port from the original Python code where `alpha = 0` is used for the Ridge fitting (note the difference in interpretation of `alpha` between glmnet and sklearns Ridge module)...

I'll be happy to accept a range of additional feature selectors as a PR if you wish and feel the need
